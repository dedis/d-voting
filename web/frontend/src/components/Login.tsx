import "./Login.css";
import { FC, useState } from "react";
import "i18n";
import { useTranslation } from "react-i18next";

import { GET_TEQ_EENDPOINT } from "./utils/ExpressEndoints";
import PropTypes from "prop-types";

const Login: FC = () => {
  const { t } = useTranslation();

  const [loginError, setLoginError] = useState();

  const handleClick = async () => {
    try {
      fetch(GET_TEQ_EENDPOINT)
        .then((resp) => {
          const json_data = resp.json();
          json_data.then((result) => {
            window.location = result["url"];
          });
        })
        .catch((error) => {
          console.log(error);
        });
    } catch (error) {
      console.log(error);
    }

    return (
      <div>{loginError === null ? <div></div> : t("errorServerDown")}</div>
    );
  };

  return (
    <div className="login-wrapper">
      <div className="login-txt">{t("loginText")}</div>
      <button id="login-button" className="login-btn" onClick={handleClick}>
        {t("login")}
      </button>
    </div>
  );
};

Login.propTypes = {
  setToken: PropTypes.func,
};

export default Login;
