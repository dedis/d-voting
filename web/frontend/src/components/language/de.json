{
  "common": {
    "Welcome to React": "Welcome to React and react-i18next",
    "about": "This is the About text"
}
}