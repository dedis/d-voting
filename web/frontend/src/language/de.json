{
  "translation": {
    "Welcome to React": "Welcome to React and react-i18next",
    "about": "This is the About text",
    "en": "üá∫üá∏ Englisch",
    "fr": "üá´üá∑ Franz√∂sisch",
    "de": "üá©üá™ Deutsch",
    "navBarStatus": "Umfragen",
    "navBarHome": "Homepage",
    "navBarCreate": "Erstellen",
    "vote": "Abstimmen",
    "forms": "Umfragen",
    "navBarResult": "Ergebnisse",
    "navBarAbout": "√úber",
    "navBarAdmin": "Admin",
    "admin": "Admin",
    "previous": "Vorherige",
    "next": "N√§chster",
    "confirmDeleteUserSciper": "Best√§tigen Sie das L√∂schen der Rolle f√ºr den Benutzer SCIPER",
    "404Title": "Seite nicht gefunden",
    "403Title": "Verbotene Seite",
    "401Title": "unbefugt Seite",
    "404Description": "Die gesuchte Seite existiert nicht.",
    "403Description": "Sie sind nicht berechtigt, auf diese Seite zuzugreifen.",
    "goHome": "Zur Homepage",
    "results": "ergebnisse",
    "showing": "zeigen",
    "saveQuestion": "Speichern",
    "addRank": "Rang hinzuf√ºgen",
    "editrank": "Rang bearbeiten",
    "removerank": "Rang entfernen",
    "addSelect": "Auswahl hinzuf√ºgen",
    "editselect": "Bearbeiten ausw√§hlen",
    "removeselect": "Auswahl entfernen",
    "addText": "Text hinzuf√ºgen",
    "edittext": "Text bearbeiten",
    "removetext": "Text entfernen",
    "subject": "Betreff",
    "choices": "Auswahlm√∂glichkeiten",
    "url": "URL",
    "additionalInfo": "Weitere Informationen",
    "answers": "Antworten",
    "enterMaxLength": "Geben Sie die MaxLength",
    "maxChoices": "Maximale Anzahl von Auswahlm√∂glichkeiten",
    "minChoices": "Minimale Anzahl von Auswahlm√∂glichkeiten",
    "enterMinN": "Eingabe der MinN",
    "enterMaxN": "Eingabe der MaxN",
    "enterRegex": "Geben Sie Ihre Regex ein",
    "mainProperties": "Wichtigste Eigenschaften",
    "additionalProperties": "Zus√§tzliche Eigenschaften",
    "removeSubject": "Betreff entfernen",
    "addSubject": "Betreff hinzuf√ºgen",
    "addQuestionrank": "Rang",
    "addQuestionselect": "Ausw√§hlen",
    "addQuestiontext": "Text",
    "importFile": "JSON Datei importieren",
    "enterSciper": "Bitte geben Sie den SCIPER des Benutzers an",
    "adminDetails": "Hinzuf√ºgen oder Entfernen von Rollen von Benutzern aus der Admin-Tabelle",
    "navBarCreateForm": "Formular erstellen",
    "homeTitle": "Willkommen auf unserer E-Voting-Plattform!",
    "homeWhatsNew": "Was gibt es Neues?",
    "homeJustShippedVersion": "Gerade ausgelieferte Version",
    "homeText": "Verwenden Sie die Navigationsleiste oben, um die gew√ºnschte Seite zu erreichen.",
    "loginText": "Sie m√ºssen sich anmelden, um auf den Inhalt von {{von}} zugreifen zu k√∂nnen.",
    "notLoggedInActionText1": "Sie m√ºssen ",
    "notLoggedInActionText2": "Anmeldung",
    "notLoggedInActionText3": " um diese Aktionen durchzuf√ºhren.",
    "loginCallback": "Wir fahren mit der Authentifizierung fort. Sie sollten  weitergeleitet werden...",
    "logout": "Abmeldung",
    "changeId": "SCIPER √§ndern",
    "changeIdTitle": "Login mit einer anderen SCIPER id",
    "changeIdDialog": "Mit welcher SCIPER id wollen Sie sich einloggen?",
    "changeIdContinue": "Login",
    "changeIdPlaceholder": "SCIPER",
    "namePlaceHolder": "Geben Sie den Namen ein",
    "addCandidate": "Einen Kandidaten hinzuf√ºgen",
    "addUser": "Benutzer hinzuf√ºgen",
    "role": "Rolle",
    "roles": "Rollen",
    "edit": "Bearbeiten",
    "nothingToAdd": "Dem ist nichts hinzuzuf√ºgen.",
    "duplicateCandidate": "Dieser Kandidat ist bereits hinzugef√ºgt worden.",
    "add": "Hinzuf√ºgen",
    "exportJSON": "Exportieren als JSON",
    "delete": "L√∂schen",
    "addVoters": "Neue W√§hlerInnen",
    "combineShares": "Aktien zusammenlegen",
    "createElec": "Formular erstellen",
    "clearForm": "Das Formular l√∂schen",
    "elecName": "Titel der Formular",
    "confirmRemovesubject": "Wollen Sie dieses Thema wirklich entfernen?",
    "confirmRemovetext": "Wollen Sie diesen Text wirklich entfernen?",
    "confirmRemoverank": "Wollen Sie diesen Rang wirklich entfernen?",
    "confirmRemoveselect": "Wollen Sie diesen Auswahl wirklich entfernen?",
    "upload": "W√§hlen Sie eine JSON-Datei von Ihrem Computer:",
    "notJson": "Die Datei muss die Erweiterung .json haben.",
    "noFile": "Keine Datei gefunden",
    "createElecDesc": "Erstellen Sie ein neues Formular, indem Sie die untenstehenden Informationen ausf√ºllen oder indem Sie",
    "uploadJSON": "Hochladen einer JSON-Datei",
    "enterMainTitleLg2": "Geben Sie den Haupttitel auf Deutsch ein    ",
    "enterMainTitleLg": "Geben Sie den Haupttitel auf Englisch ein",
    "enterMainTitleLg1": "Geben Sie den Haupttitel auf Franz√∂sisch",
    "enterSubjectTitleLg2": "Geben Sie den Titel des Themas Deutsch ein",
    "enterSubjectTitleLg": "Geben Sie den Titel des Themas Englisch ein",
    "errorAddAuth": "Beim Hinzuf√ºgen der Berechtigung zum Verwalten Ihres Formulars scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website.",
    "enterSubjectTitleLg1": "Geben Sie den Titel des Themas Franz√∂sisch ein",
    "enterTitleLg2": "Geben Sie den Titel Deutsch ein",
    "enterTitleLg": "Geben Sie den Titel Englisch ein",
    "enterTitleLg1": "Geben Sie den Titel Franz√∂sisch ein",
    "errorCandidates": "Sie m√ºssen mindestens einen Kandidaten hinzuf√ºgen!",
    "errorNewCandidate": "Sind Sie sicher, dass Sie nichts hinzuf√ºgen m√∂chten?",
    "errorRetrievingForms": "Beim Abrufen aller Formulare von unserem Server scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website. Fehler:",
    "errorRetrievingForm": "Beim Abrufen des Formulars von unserem Server scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website. Fehler: ",
    "errorRetrievingProxy": "Beim Abrufen der Adressen der Proxys von unserem Server scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website. ",
    "errorRetrievingNodes": "Beim Abrufen des Status der Knoten von unserem Server scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website. ",
    "errorRetrievingKey": "Beim Abrufen des √∂ffentlichen Schl√ºssels von unserem Server scheint ein Fehler aufgetreten zu sein. Wenden Sie sich an den Administrator dieser Website.",
    "errorServerDown": "Einer unserer Server scheint ausgefallen zu sein. Wenden Sie sich an den Administrator dieser Website.",
    "formSuccess": "Ihr Formular wurde erfolgreich abgeschickt!",
    "formFail": "Formularerstellung fehlgeschlagen!",
    "clickForm": "Klicken Sie auf den Namen des Formulars, um weitere Details anzuzeigen.",
    "noForm": "Es wurde kein Formular abgerufen!",
    "listForm": "Auf dieser Seite sind alle Formulare aufgelistet, die jemals erstellt wurden.",
    "loading": "Loading...",
    "formDetails": "Details zum Formular",
    "status": "Status",
    "startDate": "Startdatum:",
    "candidates": "Kandidaten:",
    "title": "Titel",
    "back": "Zur√ºck",
    "open": "Offen",
    "close": "Schlie√üen",
    "cancel": "Abbrechen",
    "confirm": "OK",
    "canceled": "Abgebrochen",
    "action": "Aktion",
    "login": "Login",
    "loggedIn": "Sie sind eingeloggt. ",
    "notLoggedIn": "Sie sind nicht eingeloggt.",
    "logOutSuccessful": "Logout erfolgreich.",
    "logOutError": "Abmeldung fehlgeschlagen: {{error}}",
    "confirmCloseForm": "Sind Sie sicher, dass Sie dieses Formular schlie√üen m√∂chten?",
    "confirmCancelForm": "Sind Sie sicher, dass Sie dieses Formular l√∂schen m√∂chten?",
    "confirmDeleteForm": "Sind Sie sicher, dass Sie dieses Formular l√∂schen wollen? Diese Aktion kann nicht r√ºckg√§ngig gemacht werden.",
    "createForm": "Formular erstellen",
    "statusInitial": "Erstellt",
    "statusInitializedNodes": "Knoten initialisiert",
    "initializeNode": "Knoten initialisieren",
    "initialized": "Initialisiert",
    "initializing": "Initialisierung...",
    "settingUp": "Einrichten...",
    "statusSetup": "Einrichtung",
    "setupNode": "Einrichtung Knoten",
    "statusOpen": "Offen",
    "failed": "Gescheitert",
    "dealing": "Der Umgang mit",
    "responding": "Die Antwort",
    "certifying": "Zertifizierung",
    "certified": "Zertifiziert",
    "opening": "√ñffnung...",
    "statusClose": "Geschlossen",
    "closing": "Schlie√üen...",
    "shuffling": "Schlurfen...",
    "statusShuffle": "Stimmzettel werden gemischt",
    "decrypting": "Entschl√ºsseln...",
    "statusDecrypted": "Stimmzettel entschl√ºsselt",
    "statusPubSharesSubmitted": "PubShares eingereicht",
    "combine": "Kombinieren Sie",
    "combining": "Kombinieren...",
    "statusResultAvailable": " Ergebnisse verf√ºgbar",
    "statusCancel": "Abgesagt",
    "canceling": "Absagen...",
    "errorAction": "Beim Versuch, diese Aktion auszuf√ºhren, ist ein Fehler aufgetreten. Bitte wenden Sie sich an den Administrator dieser Website. Fehler: {{Fehler}}",
    "noActionAvailable": "Es gibt nichts zu tun",
    "alreadyVoted": "Sie haben bereits abgestimmt f√ºr",
    "alreadyVoted2": "auf diesem Formular.",
    "changeVote": "Sie k√∂nnen Ihre Stimme √§ndern, indem Sie einfach eine neue Stimme abgeben.",
    "pickCandidate": "W√§hlen Sie einen Kandidaten:",
    "voteSuccess": "Ihre Stimme wurde erfolgreich abgegeben!",
    "voteSuccessful": "Stimmabgabe erfolgreich",
    "errorTitle": "Fehler",
    "actionChange": "Aktion √Ñndern",
    "notification": "Benachrichtigung",
    "successCreateForm": "Formular erfolgreich erstellt! FormID: ",
    "errorIncorrectConfSchema": "Falsche Konfiguration des Formulars, bitte f√ºllen Sie es vollst√§ndig aus: ",
    "successAddUser": "Benutzer erfolgreich hinzugef√ºgt! ",
    "errorAddUser": "Fehler beim Hinzuf√ºgen des Benutzers",
    "successRemoveUser": "Benutzer erfolgreich entfernt!",
    "errorRemoveUser": "Fehler beim Entfernen des Benutzers",
    "errorFetchingUsers": "Fehler beim Abrufen der Benutzer",
    "voteFailure": "Ihr Stimmzettel ist nicht ber√ºcksichtigt worden. Es k√∂nnte sein, dass das Formular geschlossen oder gel√∂scht wurde. Versuchen Sie, die Seite zu aktualisieren.",
    "ballotFailure": "Beim Senden Ihres Stimmzettels ist ein Fehler aufgetreten. Bitte wenden Sie sich an den Administrator dieser Website. ",
    "incompleteBallot": "Einige Antworten sind nicht vollst√§ndig.",
    "selectMin": "W√§hlen Sie {{minSelect}} {{singularPlural}}. ",
    "selectMax": "W√§hlen Sie h√∂chstens {{maxSelect}} {{singularPlural}}. ",
    "selectBetween": "W√§hlen Sie zwischen {{minSelect}} und {{maxSelect}} Antworten. ",
    "minSelectError": "Sie m√ºssen mindestens {{min}} {{singularPlural}}. ",
    "maxSelectError": "Sie k√∂nnen nicht mehr als {{max}} Antworten ausw√§hlen. ",
    "fillText": "F√ºllen Sie {{minText}} {{singularPlural}}. ",
    "minText": "F√ºllen Sie mindestens {{minText}} {{singularPlural}}. ",
    "minTextError": "Sie m√ºssen mindestens {{minText}} {{singularPlural}}. ",
    "maxTextChars": "Die Antwort muss weniger als {{maxLength}} Zeichen lang sein. ",
    "regexpCheck": "Die Antwort muss die Form haben: {{regexp}}. ",
    "singularAnswer": "Antwort",
    "pluralAnswers": "Antworten",
    "rankRange": "Die Antwort muss zwischen 1 und {{max}} liegen. ",
    "castVote": "Stimme abgeben",
    "voteExplanation": "Solange das Formular ge√∂ffnet ist, k√∂nnen Sie so oft Sie wollen abstimmen. Nur Ihre letzte Stimmabgabe wird gewertet.",
    "noVote": "Zurzeit gibt es nichts abzustimmen. ",
    "voteAllowed": "Sie k√∂nnen √ºber das/die untenstehende(n) Formular(e) abstimmen. Klicken Sie auf den Titel eines Formulars, um den entsprechenden Stimmzettel anzuzeigen und abzustimmen.",
    "displayResults": "Die Ergebnisse der unten aufgef√ºhrten Formulare sind verf√ºgbar. Klicken Sie auf einen Formulartitel, um sie aufzurufen.",
    "noResultsAvailable": "Derzeit sind keine Ergebnisse verf√ºgbar.",
    "resultExplanation1": "Die Ergebnisse f√ºr die Wahl- und Textfragen werden als Prozentsatz der Anzahl der Stimmen f√ºr einen Kandidaten geteilt durch die Anzahl der abgegebenen Stimmzettel angegeben.",
    "resultExplanation2": "Das Ergebnis der Rangfrage entspricht dem Prozentsatz der Punkte, die ein Kandidat erh√§lt. Jeder W√§hler gibt den Kandidaten Punkte, indem er sie von 1 (0 Punkte) bis N (N-1 Punkte) einstuft (niedriger ist besser).",
    "resultExplanation3": "Die Punktzahl entspricht der Summe der Punkte, die ein Kandidat erhalten hat, und wird durch die Gesamtzahl der auf allen Stimmzetteln vergebenen Punkte geteilt und zu eins subtrahiert.",
    "shuffle": "Mischen",
    "decrypt": "Entschl√ºsseln",
    "seeResult": "Siehe Ergebnisse",
    "totalNumberOfVotes": "Gesamtzahl der Stimmen : {{Abstimmungen}}",
    "notEnoughBallot": "Die Aktion ist gescheitert, weil weniger als zwei Stimmzettel abgegeben worden sind.",
    "operationFailure": "Der Vorgang ist fehlgeschlagen. Versuchen Sie, die Seite zu aktualisieren.",
    "shuffleFail": "Die Zufallsmischung ist fehlgeschlagen.",
    "voteImpossible": "Unm√∂glich abstimmen",
    "returnToFormTable": "Zur√ºck zur Formulartabelle",
    "voteImpossibleDescription": "Das Formular ist nicht mehr zur Abstimmung ge√∂ffnet.",
    "yes": "Ja",
    "no": "Nein",
    "download": "Ergebnisse im JSON-Format exportieren",
    "rowsPerPage": "Formulare pro Seite",
    "of": " von ",
    "about0": "Das folgende Diagramm zeigt das d-voting-System aus der Perspektive der Bereitstellung. Es beschreibt die Komponenten und ihr Zusammenspiel.",
    "about1": "Diese Website beherbergt die Schnittstelle zu einem Evoting-System. Dieses System f√ºhrt intelligente Vertr√§ge aus, die von einer Reihe von byzantinischen, fehlertoleranten Knoten verarbeitet werden.",
    "about2": "Wenn ein Administrator ein Formular erstellt, werden die Parameter des Formulars auf einer Blockchain gespeichert, ebenso wie jede folgende Transaktion (Schlie√üen/Abbrechen des Formulars, Abgabe einer Stimme usw.). ",
    "about3": "Bei der Erstellung des Formulars wird ein verteilter Schl√ºssel erzeugt, so dass bei der Stimmabgabe eines Benutzers seine Stimme mit dem Schl√ºssel verschl√ºsselt wird, der die Anonymit√§t der Stimme garantiert. Allerdings erzwingt das System nicht die Anonymit√§t des W√§hlers.",
    "about4": "Wenn ein Formular abgeschlossen ist, mischen die Knoten die Stimmzettel und pr√ºfen ihre Korrektheit, bevor sie die Mischung entschl√ºsseln und das Ergebnis des Formulars in einem Smart Contract ver√∂ffentlichen.",
    "end": "Das Ende",
    "save": "Speichern Sie",
    "contributors": "Our contributors",
    "addVotersDialog": "W√§hler hinzuf√ºgen",
    "addVotersDialogSuccess": "W√§hler hinzugef√ºgt",
    "votersAdded": "Folgende W√§hler wurden hinzugef√ºgt:",
    "inputAddVoters": "F√ºr jede wahlberechtigte Person, f√ºgen Sie die SCIPER Nummer auf eine neue Linie hinzu",
    "addVotersConfirm": "Hinzuf√ºgen",
    "nodeSetup": "Einrichtung des Knotens",
    "inputNodeSetup": "W√§hlen Sie den Knoten aus, auf dem die Einrichtung beginnen soll:",
    "inputProxyAddressError": "Fehler: Die Adresse eines Proxys kann nicht leer sein.",
    "node": "Knoten",
    "nodes": "Knotenpunkte",
    "DKGStatuses": "DKG-Status",
    "proxies": "Proxys",
    "all": "Alle",
    "resetFilter": "Filter zur√ºcksetzen",
    "showingNOverMOfXResults": "Zeigt {{n}}/{{m}} von {{x}} Ergebnissen.",
    "addProxy": "Proxy hinzuf√ºgen",
    "editProxy": "Bearbeiten Sie die Proxy-Adresse",
    "proxy": "Proxy",
    "confirmDeleteProxy": "Best√§tigen Sie das L√∂schen dieser Knotenadresse",
    "nodeDetails": "Hinzuf√ºgen, Bearbeiten oder Entfernen der Zuordnung zwischen einer Knotenadresse und ihrer Proxyadresse.",
    "inputNodeProxyError": "Fehler: Die Adresse des Knotens und des Proxys darf nicht leer sein.",
    "proxySuccessfullyEdited": "Die Proxy-Adresse wurde erfolgreich ge√§ndert !",
    "nodeProxySuccessfullyAdded": "Die Knoten- und Proxy-Adressen wurden erfolgreich hinzugef√ºgt!",
    "proxySuccessfullyDeleted": "Die Knoten- und Proxyadressen wurden erfolgreich gel√∂scht!",
    "addNodeProxyError": "Beim Versuch, die Knoten- und Proxyadressen hinzuzuf√ºgen, ist ein Fehler aufgetreten. Fehler:",
    "editProxyError": "Beim Versuch, die Proxy-Adresse zu bearbeiten, ist ein Fehler aufgetreten. Fehler: ",
    "removeProxyError": "Beim Versuch, die Knoten- und Proxyadressen zu entfernen, ist ein Fehler aufgetreten. Fehler:",
    "enterNodeProxy": "Bitte geben Sie die Adressen des Knotens und des Proxys ein",
    "invalidProxyError": "Fehler: Die von Ihnen eingegebene Adresse ist keine g√ºltige URL.",
    "learnMore": "Erfahren Sie mehr √ºber die D-Voting-Plattform",
    "aboutPlatform": "√úber die Plattform",
    "whatMakesUsDifferent": "Was uns auszeichnet",
    "numVotes": "Anzahl der abgegebenen Stimmzettel: {{num}}",
    "userID": "Benutzer-ID der W√§hler",
    "nodeUnreachable": "Timeout: Der Knoten ({{node}}) konnte nicht erreicht werden. ",
    "proxyUnreachable": "Timeout: die Adresse des Proxys f√ºr den Knoten ({{node}}) konnte nicht aufgel√∂st werden. ",
    "error": "Fehler: ",
    "actionLoading": "Aktion Laden...",
    "statusLoading": "Status laden...",
    "actionNotAvailable": "Aktion nicht verf√ºgbar",
    "uninitialized": "Uninitialisiert",
    "actionTextVoter1": "Das Formular ist noch nicht ge√∂ffnet, Sie k√∂nnen sp√§ter wiederkommen, um abzustimmen, sobald es ge√∂ffnet ist.",
    "actionTextVoter2": "Die Ergebnisse des Formulars liegen noch nicht vor.",
    "choice": "Auswahl",
    "logoutWarning": "Sie sind dabei, sich abzumelden. Wenn Sie gerade ein Formular ausf√ºllen, empfehlen wir Ihnen, das Formular als JSON zu exportieren, da Sie sonst Ihre √Ñnderungen verlieren. Sind Sie sicher, dass Sie fortfahren m√∂chten?",
    "continue": "Fortfahren",
    "enterHintLg1": "Geben Sie einen Hinweis auf Franz√∂sisch ein (optional)",
    "enterHintLg": "Geben Sie einen Hinweis auf Englisch ein (optional)",
    "enterHintLg2": "Geben Sie einen Hinweis auf Deutsch ein (optional)",
    "hint": "Hinweis",
    "invalidInput": "Bitte geben Sie eine Zahl zwischen 1 und {{max}} ein.",
    "footerCopyright": "DEDIS LAB & C4DT -",
    "footerUnknown": "?",
    "footerVersion": "version:",
    "footerBuild": "build:",
    "footerBuildTime": "in:",
    "voteNotVoter": "W√§hlen nicht erlaubt.",
    "voteNotVoterDescription": "Sie sind nicht wahlberechtigt in dieser Wahl. Falls Sie denken, dass ein Fehler vorliegt, wenden Sie sich bitte an die verantwortliche Stelle.",
    "addVotersLoading": "W√§hlerInnen werden hinzugef√ºgt...",
    "sciperNaN": "'{{sciperStr}}' ist keine Zahl; ",
    "sciperOutOfRange": "{{sciper}} ist nicht in dem erlaubten Bereich (100000-999999); ",
    "invalidScipersFound": "Ung√ºltige SCIPERs wurden gefunden. Es wurde keine Anfrage gesendet. Bitte korrigieren Sie folgende Fehler: {{sciperErrs}}"
  }
}
