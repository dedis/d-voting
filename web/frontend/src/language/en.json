{
  "common": {
    "Welcome to React": "Welcome to React and react-i18next",
    "navBarHome": "Home",
    "navBarCreate": "Create an election",
    "navBarStatus": "Election status",
    "navBarVote": "Vote",
    "navBarResult": "Results",
    "navBarAbout": "About",
    "homeTitle": "Welcome to our e-voting platform!",
    "homeText": "Use the navigation bar above to reach the the page you want.",
    "loginText": "You need to login to access the content of the website.",
    "elecName": "Election title",
    "namePlaceHolder": "Enter the name",
    "addCandidate": "Add a candidate",
    "addCandPlaceHolder": "candidate's name",
    "nothingToAdd": "There is nothing to add.",
    "duplicateCandidate": "This candidate has already been added.",
    "add": "Add",
    "delete": "Delete",
    "createElec": "Create election",
    "upload": "Choose a json file from your computer:",
    "notJson": "The file needs to have the .json extension.",
    "noFile": "No file found",
    "create": "Create a new election by filling out the information below or by uploading a json file",
    "errorCandidates": "You must add at least one candidate!",
    "errorNewCandidate": "Are you sure you don't want to add ",
    "errorRetrievingElection ": "An error seems to have occured while retrieving all the elections from our server. Contact the administrator of this website.",
    "errorRetrievingKey": "An error seems to have occured while retrieving the public key from our server. Contact the administrator of this website.",
    "errorServerDown ": "One of our servers seems to be down. Contact the administrator of this website.",
    "electionSuccess":"Your election was successfully submitted!",
    "electionFail": "Election creation failed!",
    "clickElection": "Click on the election name to display additional details.",
    "noElection":"No election were retrieved!",
    "listElection": "This page lists all the elections that have ever been created.",
    "loading": "Loading...",
    "electionDetails": "Election details",
    "status": "Status",
    "startDate": "Start date:",
    "candidates": "Candidates:",
    "title ": "Title",
    "back": "Back",
    "close": "Close",
    "cancel": "Cancel",
    "action ": "Action",
    "login": "Login",
    "confirmCloseElection": "Are you sure you want to close this election?",
    "confirmCancelElection": "Are you sure you want to cancel this election?",
    "statusOpen": "Open",
    "statusClose":"Closed",
    "statusCancel": "Canceled",
    "statusShuffle ": "Ballots shuffled",
    "shuffleOnGoing": "Shuffling...",
    "decryptOnGoing": "Decrypting...",
    "resultsAvailable": " Results available",
    "alreadyVoted": "You have already voted for",
    "alreadyVoted2": "on this election.",
    "changeVote":  "You can change your vote by simply casting a new vote.",
    "pickCandidate": "Pick a candidate:",
    "voteSuccess": "Your vote was successfully submitted!",
    "voteFailure ": "Your ballot hasn't been taken into account. It might be that the election has been closed or cancelled. Try refreshing the page.",
    "noCandidate": "You need to select a candidate.",
    "castVote": "Cast vote",
    "noVote": "There is currently nothing to vote on.",
    "voteAllowed": "You are allowed to vote on the election(s) below. Click on an election title to display its ballot and vote.",
    "displayResults": "The results of the election(s) listed below are available. Click on an election title to access them.",
    "noResultsAvailable": "There is currently no available results.",
    "shuffle": "Shuffle",
    "decrypt": "Decrypt",
    "seeResult ": "See results",
    "notEnoughBallot": "The operation failed because less than two ballots have been casted.",
    "operationFailure": "The operation failed. Try refreshing the page.",
    "shuffleFail ": "The shuffle operation failed.",
    "voteImpossible": "The election is not open for voting anymore.",
    "yes": "Yes",
    "no": "No",
    "download": "Export results in JSON format",
    "rowsPerPage ": "Elections per page",
    "of": " of ",
    "about1": "This website hosts the interface of an evoting system. This system runs smart contracts, handled by a set of Byzantine fault-tolerant nodes.",
    "about2": "When an administrator creates an election, the election parameters are saved on a blockchain and so are every following transaction (closing/cancelling election, casting a vote,...). ",
    "about3": "A distributed key is generated at election creation time so that when a user votes, his/her vote is encrypted with the key guarantying the anonymity of the vote. However the system doesn't enforce the anonymity of the voter.",
    "about4": "When an election is closed, the nodes shuffle the ballots and check its correctness before decrypting the shuffle and publish the result of the election on a smart contract.",
    "end": "The end"
  }
}