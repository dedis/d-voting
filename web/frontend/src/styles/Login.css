.login-wrapper{
    display: flex;
    flex-direction: column;
    align-items: center;


}
.login-txt{
    display: flex;
    padding-top: 5%;
    padding-bottom: 10%;

}
.login-btn{
    display: flex;

}