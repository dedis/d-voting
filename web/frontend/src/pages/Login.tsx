import { FC, useState } from "react";
import { useTranslation } from "react-i18next";
import PropTypes from "prop-types";

import { GET_TEQ_EENDPOINT } from "../components/utils/ExpressEndoints";
import "../styles/Login.css";

const Login: FC = () => {
  const { t } = useTranslation();

  const [loginError, setLoginError] = useState();

  const handleClick = async () => {
    try {
      fetch(GET_TEQ_EENDPOINT)
        .then((resp) => {
          const json_data = resp.json();
          json_data.then((result) => {
            window.location = result["url"];
          });
        })
        .catch((error) => {
          console.log(error);
        });
    } catch (error) {
      console.log(error);
    }

    return (
      <div>{loginError === null ? <div></div> : t("errorServerDown")}</div>
    );
  };

  return (
    <div className="login-wrapper">
      <div className="login-txt">{t("loginText")}</div>
      <button id="login-button" className="login-btn" onClick={handleClick}>
        {t("login")}
      </button>
    </div>
  );
};

Login.propTypes = {
  setToken: PropTypes.func,
};

export default Login;
