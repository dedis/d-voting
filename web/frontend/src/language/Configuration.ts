import { use } from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';

import de from './de.json';
import en from './en.json';
import fr from './fr.json';

const resources = {
  de,
  en,
  fr,
};

export const availableLanguages = Object.keys(resources);

use(initReactI18next)
  .use(LanguageDetector)
  .init({
    resources,
    fallbackLng: ['en', 'fr', 'de'],
    debug: true,
    interpolation: {
      escapeValue: false, // react already safe from xss => https://www.i18next.com/translation-function/interpolation#unescape
    },
  });
