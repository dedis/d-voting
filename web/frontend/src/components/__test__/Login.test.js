/*global , describe, it, expect*/
/*eslint no-undef: "error"*/

import React from 'react';
import Login from '../login/Login';
import renderer from 'react-test-renderer';
import {LanguageContext} from '../language/LanguageContext'

describe('Login', ()=> {
    it('should render the Login Component correctly', () => {  
        const component = renderer.create(<LanguageContext.Provider value = {['en',]}><Login /></LanguageContext.Provider>);
        let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    });
})