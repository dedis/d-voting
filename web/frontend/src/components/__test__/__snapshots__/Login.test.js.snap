// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render the Login Component correctly 1`] = `
<div
  className="login-wrapper"
>
  <div
    className="login-txt"
  >
    You need to login to access the content of the website.
  </div>
  <button
    className="login-btn"
    id="login-button"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
